### DISCUSSION TITLE
[Question] Button as prop에 컴포넌트 주입시 disabled 처리

### DISCUSSION THREAD

**Discussion by jy-son** [wanteddev/montage-web#69] [Q&A] [answered]
### 요약

Button 컴포넌트에서 as prop으로 react-router-dom의 Link를 주입하고 있습니다.
disabled prop이 true인 경우 동작은 안하지만 UI가 반영이 되지 않습니다 😢 

### 상세

```ts
const Demo = () => {
  return (
    <Button disabled>텍스트</Button>
  )
}
```

<img width="939" alt="image" src="https://github.com/user-attachments/assets/404e36cf-2072-4aa2-89d2-592333b7de77">

```ts
const Demo = () => {
  return (
    <Button as="a" disabled>텍스트</Button>
  )
}
```

<img width="985" alt="image" src="https://github.com/user-attachments/assets/f516403a-6e43-4a24-8ad3-75dd4411861b">

현재 임의로 `&[aria-disabled='true']` 선택자로 수동으로 적용하고 있습니다.

**Comment by Sh031224** [comment, marked as answer]
a 태그 자체적으로는 disabled를 지원하지 않기 때문에 발생하는 것 같습니다.
aria-disabled로 처리할 수 있도록 수정하겠습니다.

**Reply by jy-son** [reply]
disabled 기준으로 스타일이 적용되어 있었군요.
해당 부분 참고하고 버전 업데이트 때 대응할게요!
급한 이슈는 아닙니다!
